Source/ASDisplayNode.mm

```diff
@@ -433,6 +433,9 @@ - (void)dealloc
 
   // TODO: Remove this? If supernode isn't already nil, this method isn't dealloc-safe anyway.
   [self _setSupernode:nil];
+  
+  ASPerformMainThreadDeallocation(&_interfaceStateDelegates);
```

Comment:
I don't think this is necessary. If the node is gone, `_interfaceStateDelegates` can be safely deallocated as well, especially because it only stores weak references.